File: 3214-maximize-area-of-square-hole-in-grid/maximize-area-of-square-hole-in-grid.java
import java.util.*;

class Solution {
    public int maximizeSquareHoleArea(int n, int m, int[] hBars, int[] vBars) {
        int maxH = longestConsecutive(hBars);
        int maxV = longestConsecutive(vBars);
        
        // side length of square hole
        int side = Math.min(maxH, maxV) + 1;
        
        return side * side;
    }
    
    private int longestConsecutive(int[] bars) {
        if (bars.length == 0) return 0;
        Arrays.sort(bars);
        int maxLen = 1, currLen = 1;
        
        for (int i = 1; i < bars.length; i++) {
            if (bars[i] == bars[i-1] + 1) {
                currLen++;
            } else {
                currLen = 1;
            }
            maxLen = Math.max(maxLen, currLen);
        }
        return maxLen;
    }
}

File: 3214-maximize-area-of-square-hole-in-grid/maximize-area-of-square-hole-in-grid.py
from typing import List

class Solution:
    def maximizeSquareHoleArea(self, n: int, m: int, hBars: List[int], vBars: List[int]) -> int:
        def longest_consecutive(bars: List[int]) -> int:
            if not bars:
                return 0
            bars.sort()
            max_len = 1
            curr_len = 1
            for i in range(1, len(bars)):
                if bars[i] == bars[i-1] + 1:
                    curr_len += 1
                else:
                    curr_len = 1
                max_len = max(max_len, curr_len)
            return max_len
        
        maxH = longest_consecutive(hBars)
        maxV = longest_consecutive(vBars)
        
       
        side = min(maxH, maxV) + 1
        return side * side
